### PR TITLE
#7814 - Daylight savings time leading to incorrect season for tariff calculation

### LINKED ISSUE #7814: Daylight savings time adding incorrect month calculation in time of use tariff
Issue overview
--------------
The UtilityCost:Tariff object with a time of use period schedule and a season schedule that both start the summer on June 1 are showing summer tariff calculations starting in May. This file has a RunPeriodControl:DaylightSavingTime which starts daylight savings saving in March. When the RunPeriodControl:DaylightSavingTime is removed, the summer calculations are started in June when it is supposed to be. 

### Details
Some additional details for this issue (if relevant):
 - EnergyPlus 9.2.0
 - https://unmethours.com/question/42857/utility-tariff-not-calculating-midpeak-the-first-month/

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Defect file added (devsupport)
 - [ ] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

### PR DESCRIPTION
**Pull request by jmarrec**
Pull request overview
---------------------
 - Fixes #7814 
 - Daylight savings time leading to incorrect season for tariff calculation

I added a MCVE on EnergyPlusDevSupport: [7814_mcve.idf](https://github.com/NREL/EnergyPlusDevSupport/blob/master/DefectFiles/7000s/7814/7814_mcve.idf)

And I performed an analysis of before / after fix in a jupyter notebook, but also saved the result as an HTML file for easy viewing in any browser locally: [Analyze_MCVE.htm](https://github.com/NREL/EnergyPlusDevSupport/blob/master/DefectFiles/7000s/7814/Analyze_MCVE.html)

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [x] Perform a Code Review on GitHub
 - [x] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [x] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [x] CI status: all green or justified
 - [x] Check that performance is not impacted (CI Linux results include performance check)
 - [x] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

File: src/EnergyPlus/EconomicTariff.cc [hunk 1]
```diff
@@ -2785,7 +2785,13 @@ namespace EconomicTariff {
...
                         curMonth = GetCurrentScheduleValue(tariff(iTariff).monthSchIndex);
                     } else {
-                        curMonth = Month; // from DataEnvironment
+                        // #7814 - Have to carefull with DST. tariff::seasonForMonth is overwritten at each timestep, and only the last value is
+                        // retained, so make sure to capture the right one
+                        if ((DataGlobals::HourOfDay + DataEnvironment::DSTIndicator) <= 24) {
+                            curMonth = DataEnvironment::Month;
+                        } else {
+                            curMonth = DataEnvironment::MonthTomorrow;
+                        }
```
Comment (jmarrec) [inline src/EnergyPlus/EconomicTariff.cc:2794]: Fix is here (quite simple in the end)

File: tst/EnergyPlus/unit/EconomicTariff.unit.cc [hunk 2]
```diff
@@ -447,3 +453,227 @@ TEST_F(EnergyPlusFixture, EconomicTariff_LEEDtariffReporting_Test)
...
+    EXPECT_EQ(3.0, ScheduleManager::GetCurrentScheduleValue(tariff(1).seasonSchIndex));
+
+    ExteriorEnergyUse::ManageExteriorEnergyUse(state.exteriorEnergyUse);
+
+    EXPECT_EQ(1000.0, state.exteriorEnergyUse.ExteriorLights(1).Power);
+    EXPECT_EQ(state.exteriorEnergyUse.ExteriorLights(1).Power * DataGlobals::TimeStepZoneSec, state.exteriorEnergyUse.ExteriorLights(1).CurrentUse);
+
+    // This Should now call GatherForEconomics
+    EconomicTariff::UpdateUtilityBills();
+    EXPECT_EQ(1, EconomicTariff::tariff(1).seasonForMonth(5));
```
Comment (jmarrec) [inline tst/EnergyPlus/unit/EconomicTariff.unit.cc:676]: Same as the defect file for #7814 , Will fail without fix

**Comment by JasonGlazer** [comment]
Assuming the CI comes back clean and performance looks good, this is ready to merge.

**Comment by JasonGlazer** [comment]
The AUD, RDD and MTD diffs seem unrelated but I see this branch has fallen behind so I will remerge with develop.

**Comment by JasonGlazer** [comment]
@jmarrec I just merged this, thanks for tracking down and fixing this issue.

**Comment by jmarrec** [comment]
Thanks for the review!

**Pull request merged**